src/main/java/net/einsteinsci/betterbeginnings/register/recipe/AdvancedRecipe.java

```diff
@@ -182,19 +194,18 @@ public boolean checkMatchMostly(InventoryCrafting crafting, int width, int heigh
 
 				ItemStack craftingStackInQuestion = crafting.getStackInRowAndColumn(k, l);
 
-				if (craftingStackInQuestion != null || neededCraftingStack != null)
+				if (craftingStackInQuestion != null && neededCraftingStack != null)
 				{
-					// If one is null but not the other
-					if (craftingStackInQuestion == null || neededCraftingStack == null)
-					{
-						return false;
-					}
-
 					if (!neededCraftingStack.matches(craftingStackInQuestion))
 					{
 						return false;
 					}
 				}
+				else if ((craftingStackInQuestion == null) ^ (neededCraftingStack == null))
```

Comment:
Understandable, I had to do some searching to double check, myself. It was performing as expected though so I was pretty sure it was good.